Update Dapp and Wallet examples with Elrond RPCs

radumojic opened the pull request:
We are working on enabling the WalletConnect V2 support accross the Elrond ecosystem and the added examples here would be beneficial to any developer looking to integrate Elrond support too

**CAIP-2 Specs:**
namespace: `elrond`
reference: `1` for Mainnet, `T` for Testnet, `D` for Devnet ( same as Elrond chainID )

**Test Cases:**
```
# Elrond Mainnet
elrond:1:erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l

# Elrond Devnet
elrond:D:erd1devnet6uy8xjusvusfy3q83qadfhwrtty5fwa8ceh9cl60q2p6ysra7aaa

# Elrond Testnet
elrond:T:erd1sea63y47u569ns3x5mqjf4vnygn9whkk7p6ry4rfpqyd6rd5addqyd9lf2
```


**Changelist:**

`react-dapp-v2` & `react-wallet-v2`
- Add Elrond Mainnet and Devnet ( test ) Chain Data
- Add Elrond RPC Methods
	- sign Transaction
	- sign a Message
	- sign a batch of Transactions

`react-wallet-v2` 
- Add Elrond test Wallet Mnemonic in Settings	

---

You can check out the changes here:

**Wallet:**
[https://wc2-wallet.vercel.app/](https://wc2-wallet.vercel.app/)

**dApp:**
[https://wc2-dapp.vercel.app/](https://wc2-dapp.vercel.app/)

---

**Useful Links:**
Website: https://elrond.com/
Docs: https://docs.elrond.com/

bkrem commented:
Please remove the `.DS_Store` dotfile that was added here (can't comment it directly). Feel free to add it to `.gitignore` if that makes things easier, otherwise I can remedy that later 👍

bkrem reviewed (changes requested):
Some minor cleanup needed + a question around next config, but looking great overall, thanks for adding Elrond to the canonical v2 examples 🙌

bkrem commented on dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx:
```diff
@@ -814,6 +830,169 @@ export function JsonRpcContextProvider({
+            },
+          });
+
+          const valid = true; // Implement transaction signature validity check once it is implemented in erdjs
```
Is this still current, i.e. is this available now as part of the `erdjs` lib?

If not is there another, more low-level way to validate the returned signatures instead of defaulting to `true`?

radumojic replied:
Added the validity check for Elrond transacions too

radumojic commented:
> Please remove the `.DS_Store` dotfiles that were added here (can't comment it directly). Feel free to add it to `.gitignore` if that makes things easier, otherwise I can remedy that later 👍

removed .DS_Store files outside .gitignores

radumojic commented on dapps/react-dapp-v2/next.config.js:
```diff
@@ -3,6 +3,14 @@ const nextConfig = {
   reactStrictMode: true,
   swcMinify: true,
   distDir: "build",
+  webpack(config) {
```
Hey, this config is currently needed because webpack no longer includes polyfills for node.js core modules by default

The fs module is used in a dependency of `@elrondnetwork/erdjs-walletcore` (`@elrondnetwork/bls-wasm`) that's not currently used in code, so there's no functionality impact. 
We are working on extracting the packages causing the issue to separate repositories, but this might take a while.

We will update the config file once we have the new version.

bkrem replied:
Gotcha, thanks for clarifying 👍

radumojic commented:
> Some minor cleanup needed + a question around next config, but looking great overall, thanks for adding Elrond to the canonical v2 examples 🙌

Hello, I've addressed the feedback received, please let me know if there's anything else I can help with 👍

bkrem commented:
Thanks for addressing those comments @radumojic 💯 

Was doing some final testing on the deployments, establishing testnet session and signing works as expected, but for Elrond mainnet I'm getting what looks like Ethers errors which throw the whole dapp and require me to call `localStorage.clear()` to recover it:

<img width="556" alt="Screenshot 2022-10-28 at 10 29 29" src="https://user-images.githubusercontent.com/8663099/198542345-373d119a-4bec-411f-bd6f-7d0518268bc2.png">

Repro:

1. Toggle dapp to mainnets
2. Select Elrond
3. Establish a session with the wallet
4. Dapp throws with above error

Once this is addressed we should be good to go 👍

radumojic commented:
> 

hello, the issue was addressed [here](https://github.com/WalletConnect/web-examples/blob/78b9f76c2c744f646bbc2b55d3949ebab2d4302c/dapps/react-dapp-v2/src/helpers/api.ts#L109) i've added an extra check for the chain `namespace` before trying to fetch the `eth_getBalance` based on chainId, since chains with different namespaces might have the same chainId ( in case of Elrond Mainnet the '1' chainId was is the same as Ethereum's Mainnet )

bkrem commented:
> hello, the issue was addressed [here](https://github.com/WalletConnect/web-examples/blob/78b9f76c2c744f646bbc2b55d3949ebab2d4302c/dapps/react-dapp-v2/src/helpers/api.ts#L109) i've added an extra check for the chain namespace before trying to fetch the eth_getBalance based on chainId, since chains with different namespaces might have the same chainId ( in case of Elrond Mainnet the '1' chainId was is the same as Ethereum's Mainnet )

Ah wow so that was a bad implicit assumption on the dapp side, sorry! Thank you for handling that 🙏

bkrem reviewed (approved):
Great work, thanks again for adding Elrond! 💯
